crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -330,8 +332,42 @@ impl SearchPaths {
         &self.typeshed_versions
     }
 
-    pub fn python_version(&self) -> Option<&PythonVersionWithSource> {
-        self.python_version.as_ref()
+    pub fn python_version(&self) -> Option<Cow<PythonVersionWithSource>> {
+        if let Some(version) = self.python_version_from_metadata.as_ref() {
+            return Some(Cow::Borrowed(version));
+        }
+
+        if cfg!(windows) {
+            // The path to `site-packages` on Unix is
+            // `<sys.prefix>/lib/pythonX.Y/site-packages`,
+            // but on Windows it's `<sys.prefix>/Lib/site-packages`.
+            return None;
+        }
+
+        let primary_site_packages = self.site_packages.first()?.as_system_path()?;
+
+        let mut site_packages_ancestor_components = primary_site_packages
+            .components()
+            .rev()
+            .skip(1)
+            .map(|c| c.as_str());
+
+        let parent_component = site_packages_ancestor_components.next()?;
```

Comment:
This won't strictly be the parent component. E.g if you have `D:\`, then the second last component is the ROOT component (or dir?). 

Should this be using `ancestor` instead of `components`?

Reply:
> E.g if you have `D:\`, then the second last component is the ROOT component (or dir?).

I think we can be confident that that won't be the case here. We wouldn't have succeeded in resolving the `site-packages` directory unless that directory was the grandchild of the `sys.prefix` path. Should I add a comment?

> Should this be using `ancestor` instead of `components`?

Hmm, I think that would be more fiddly to deal with? `ancestors` iterates over the ancestor paths (`<path/to/site-packages>.ancestors()` -> `[<path/to/site-packages>, <path/to>, <path>]`), so I'd have to call `.file_name()` on each path yielded by the method to be able to inspect the information I'm interested in

Reply:
Oh right, you only want components. A short comment might be helpful althought it might also be fine the way it is. Up to you